src/utils/evm/topUp.ts

```diff
@@ -0,0 +1,61 @@
+import { BigNumber, Wallet, providers, utils } from 'ethers'
+import { BalanceConfig, RpcEndpoints } from '../../types'
+import { EVM_BLOCK_CONFIRMATIONS } from '../../constants'
+
+export async function checkEVMBalanceAndTopUp(
```

Comment:
Can you just call this this topUP and make a separate function called checkEVMBalance that is called inside it. 
 I don't think the naming has to be that verbose.